### PR TITLE
Clinical Lab chart with extra Y axis, missing tests

### PR DESCRIPTION
**Pull request by larslemos**
Added the required features of providing graph labels that respond to clicks,
changed interpolated graphs into stepped lines,
and Y axis to be used to measure the weight line.
Missing straight lines to mark the start date of the TARV and tests for all the features.

### CODE REVIEW

File: app/clinic/controllers/patientChartController.js [hunk 1]
```diff
@@ -1,78 +1,115 @@
...
+          "e1da52ba-1d5f-11e0-b929-000c29ad1d07"];
+
+        var series = [$filter('translate')('CLINIC_PATIENT_WEIGHT'), $filter('translate')('CLINIC_PATIENT_HEIGHT'),
+          $filter('translate')('CLINIC_PATIENT_BMI')];
+        var name = "bmi";
+
+        var adultFollowupEncounterUuid = "e278f956-1d5f-11e0-b929-000c29ad1d07";//TODO: create in configuration file
+        var childFollowupEncounterUuid = "e278fce4-1d5f-11e0-b929-000c29ad1d07";//TODO: create in configuration file
+
+        //TODO: Fix, show diffenrent graphs for adult and child
```
Comment (psbrandt) [inline app/clinic/controllers/patientChartController.js]: Does this `TODO` also exist as a Trello card?

File: app/clinic/controllers/patientChartController.js [hunk 2]
```diff
@@ -1,78 +1,115 @@
...
+    function ($scope, $filter, $stateParams, encounterService,
+              observationsService) {
+      var patientUuid;
+
+      (function () {
+        patientUuid = $stateParams.patientUuid;
+
+      })();
+
+      $scope.initImc = function () {
```
Comment (ningosi) [inline app/clinic/controllers/patientChartController.js]: I think if this concepts would be given names describing what they are, would be helpful for someone who would like to check those concepts in the dictionary using the openmrs UI. For example if we could have something like 
var CD4 = 'uuid';
var WEIGHT = 'uuid';
var graphConcepts = [CD4, WIGHT, BMI]; for example.

Though it is NOT a `show stopper` for now

File: app/clinic/controllers/patientChartController.js (hunk 1 above)
Comment (larslemos) [inline app/clinic/controllers/patientChartController.js]: @psbrandt Thank you. Will add the card to the board

File: app/clinic/controllers/patientChartController.js (hunk 2 above)
Comment (larslemos) [inline app/clinic/controllers/patientChartController.js]: @ningosi Thank you. There is a plan to add that feature and also the ability to increase the number of concepts display in the graph on request of the clinic. Will add that to a card

File: app/clinic/controllers/patientChartController.js [hunk 3]
```diff
@@ -1,78 +1,115 @@
...
+        encounterService.getEncountersForEncounterType(patientUuid,
+          ($scope.patient.age.years >= 15) ? adultFollowupEncounterUuid : childFollowupEncounterUuid)
+          .success(function (data) {
+            filterObs(data, concepts, series, name);
+          });
+      };
+
+      $scope.initLabChart = function () {
+        var labEncounterUuid = "e2790f68-1d5f-11e0-b929-000c29ad1d07";
+        var concepts = ["e1e68f26-1d5f-11e0-b929-000c29ad1d07", "e1d48fba-1d5f-11e0-b929-000c29ad1d07",
```
Comment (valvijo) [inline app/clinic/controllers/patientChartController.js]: Can we move these UUIDs to `constants.js`? The same UUIDs can be used in many places.

Comment (larslemos) [inline app/clinic/controllers/patientChartController.js]: @valvijo Yes, good idea. Will add to a TODO card. Thank you.

**Pull request merged**